### ISSUE TITLE
(Bug report) Copy and Paste not working, only Text is pasted not Images

### ISSUE THREAD

**Issue by megabugman** [TriliumNext/Notes#521]
### TriliumNext Version

0.90.9 beta

### What operating system are you using?

Windows

### What is your setup?

Local (no sync)

### Operating System Version

Windows 10 Version 10.0.19045

### Description

Select text and pictures, copy to clipboard, paste into another application only pastes the text not the pictures:
![2024-10-25_19h34_35-00 00 00 000-00 00 07 601](https://github.com/user-attachments/assets/fd86fcd0-4407-4f24-8c1a-ee7a34e8ef06)

How to paste images as well?

**Comment by Onionbai** [comment]
In trilium, images are stored in internal database, if you directly copy an image, that is copying the internal link to this image. To get this image, first double click on the image, you'll see the the image in attachment view, and then right click on the image, you will see two options, copy reference to clipboard and copy image to clipboard, click on the latter, then you have the image in your clipboard.
![image](https://github.com/user-attachments/assets/df37305a-3860-48de-a754-ee9fbd3bf15c)

**Comment by perfectra1n** [comment]
Yeah, this is a feature (definitely can be an annoying one). #452 should appropriately copy the image to your clipboard when following what @Onionbai said. However, perhaps we should evaluate if we would like to override this longstanding "feature"? @eliandoran what do you think?

**Comment by bwat47** [comment]
Perhaps there could be a preference in the settings for the default image copy behavior? E.g. default to copying the internal link (current behavior) or default to copying the actual image.

**Comment by eliandoran** [comment]
> Perhaps there could be a preference in the settings for the default image copy behavior? E.g. default to copying the internal link (current behavior) or default to copying the actual image.

The problem with this approach is that copy-paste is also used internally when copying text between notes. There it makes sense to use the internal link.

A more complicated solution would involve having a different paste behavior in-app vs out-of-app. I guess this could be achieved via custom clipboard categories, but it would involve quite a bit of work.

**Comment by Onionbai** [comment]
@Nriver has now made it possible to copy image into clipboard, using [image-zoom-widget](https://github.com/Nriver/image-zoom-widget) extension. 
If you want to directly copy images and texts in your notes, another way is to share it and copy in html view, that is a better way, I think.
![Image](https://github.com/user-attachments/assets/97900bd2-4b2f-410a-aa58-61056764e1ff)

**Comment by Onionbai** [comment]
Share notes, well, I think you should read the wiki first, it is located in every note's basic properties -> Shared button. Share option does not require a server to be available. In [wiki](https://triliumnext.github.io/Docs/) -> Advanced Usage -> Sharing, there is the explanation for sharing notes

![Image](https://github.com/user-attachments/assets/bc8e490a-b707-4ca2-b9bb-d99bae8974b2)

![Image](https://github.com/user-attachments/assets/7a3c5381-d13e-4d54-9be7-b9ad0b0f91da)

**Comment by bwat47** [comment]
Thanks, I completely missed looking on the basic properties tab :)

Sharing and copying from the page in the web browser helps a little, but I'm still seeing some odd behavior pasting images into external applications.

If I just do ctrl + a to select everything (or highlight both text and images with the mouse and copy) and then try to paste in an external application, the image is broken. For example pasting into Outlook in the below video:

https://github.com/user-attachments/assets/e085c9c9-8bdd-4927-9271-8ed36dfe8f3f

My notes have various canned replies (some of which have inline images). My main use case here is to copy both text and images all at once into an email compose window (without having to manually copy each image).

NOTE: This doesn't happen with all applications. In some applications (Outlook Win32, MS Word, Libreoffice Writer) the images do paste without issue (as seen when pasting into Word in my above video). 

In other applications (eM Client or the web based Outlook), the pasted image is broken (unless I copy the image on it's own with right click | copy image).

**Comment by bwat47** [comment]
I noticed that with Obsidian, if I used either the Copy Document as HTML plugin or the pandoc plugin and export the note to HTML, the inline images paste fine into all applications

the main difference seems to be that those obsidian plugins embed the images by converting them to base64

with the trilium share page, the images are served from a local http URL (e.g. http://127.0.0.1:37840/share/api/attachments/xxxxxxxxxx/image/image.png). 

I suspect outlook web and certain other apps are refusing to embed images from plain http URLs

**Comment by bwat47** [comment]
Looks like that theory wasn't quite in the mark

I set up a trilium server instance on my home server, using my cloudflare domain so it's served with https and valid certifcate, but the behavior doesn't change when pasting into outlook even when using the share URL with everything served via https
